chat-widget/src/components/footerstateful/downloadtranscriptstateful/DownloadTranscriptStateful.tsx

```diff
@@ -167,8 +169,14 @@ const beautifyChatTranscripts = (chatTranscripts: string, renderMarkDown?: (tran
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 export const downloadTranscript = async (chatSDK: any, downloadTranscriptProps: IDownloadTranscriptProps, state?: ILiveChatWidgetContext) => {
-    // Need to keep existing live chat context for scenarios when trnascript is downloaded after endchat
-    const liveChatContext = state?.domainStates?.liveChatContext;
+    const inMemoryState = executeReducer(state as ILiveChatWidgetContext, { type: LiveChatWidgetActionType.GET_IN_MEMORY_STATE, payload: null });
+
+    // Need to keep existing live chat context for scenarios when transcript is downloaded after endchat
```

Comment:
why not to return directly from inMemory if is not null?

Reply:
It's because the issue only happens specifically on `Post Chat Survey Link Mode` where `liveChatContext` is undefined in the `state`. If  `liveChatContext` is not defined from the state, then we'd try it from the `inMemoryState`